src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationReleaseOrderService.cs

```diff
@@ -0,0 +1,226 @@
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Release = GovUk.Education.ExploreEducationStatistics.Content.Model.Release;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class PublicationReleaseOrderService : IPublicationReleaseOrderService
+    {
+        private readonly ContentDbContext _context;
+
+        public PublicationReleaseOrderService(
+            ContentDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task CreateForCreateLegacyRelease(Guid publicationId, Guid releaseId)
+        {
+            var publication = await GetPublication(publicationId);
+
+            if (releaseId == Guid.Empty)
+            {
+                throw new ArgumentNullException(nameof(releaseId));
+            }
+
+            publication.ReleaseOrders.Add(new()
+            {
+                ReleaseId = releaseId,
+                IsLegacy = true,
+                IsDraft = false,
+                Order = publication.ReleaseOrders.Count + 1 // Add to the top of the list rather than presume the intended position
+            });
+
+            _context.Update(publication);
+        }
+
+        public async Task DeleteForDeleteLegacyRelease(Guid publicationId, Guid releaseId)
+        {
+            var publication = await GetPublication(publicationId);
+
+            if (releaseId == Guid.Empty)
+            {
+                throw new ArgumentNullException(nameof(releaseId));
+            }
```

Comment:
Add the `#nullable enable` pragma to this class and then it will be clear that the id argument is a non-nullable reference type.

I think it's overkill to check for a `null` or `Guid.Empty` value here and we don't do this level of argument checking elsewhere.